### ISSUE TITLE
it seems a network exception,it seems just broke one network packet can cause the exception？

### ISSUE THREAD

**Issue by 2jiangjiang** [onimp/oni_multiplayer#283]
InvalidOperationException: An attempt was made to transition a task to a final state when it had already completed.

System.Threading.Tasks.TaskCompletionSource`1[TResult].SetResult (TResult result) (at <695d1cc93cca45069c528c15c9fdd749>:0)
MultiplayerMod.Platform.Steam.Network.SteamServer.ConnectedToSteamCallback () (at <9d5476f5734f4b7186d16fe17f585d05>:0)
MultiplayerMod.Platform.Steam.Network.SteamServer.<Initialize>b__44_0 (Steamworks.SteamServersConnected_t _) (at <9d5476f5734f4b7186d16fe17f585d05>:0)
Steamworks.Callback`1[T].OnRunCallback (System.IntPtr pvParam) (at <265662ccba5f4b45b650e9f26e9148fe>:0)
UnityEngine.Debug:LogException(Exception)
Steamworks.CallbackDispatcher:ExceptionHandler(Exception)
Steamworks.Callback`1:OnRunCallback(IntPtr)
Steamworks.CallbackDispatcher:RunFrame(Boolean)
Steamworks.GameServer:RunCallbacks()
MultiplayerMod.Platform.Steam.Network.SteamServer:Tick()
MultiplayerMod.Platform.Steam.Network.Components.SteamServerComponent:Update()

Build: U49-577063-V

**Comment by polycone** [comment]
Yes, that's it, we rely on Steam stable connection for now. We plan to address it in the QoL 2 release.
